### PR TITLE
Moved productivity estimator method from WorkUnit to DefaultWorkEstimator

### PR DESCRIPTION
**Pull request by vanoha**

### CODE REVIEW

File: sampo/schemas/time_estimator.py [hunk 1]
```diff
@@ -1,21 +1,108 @@
...
-    def estimate_time(self, work_name: str, work_volume: float, resources: WorkerContractorPool) -> Time:
+    def estimate_time(self, work_unit: WorkUnit, worker_list: list[Worker]):
         ...
 
 
 # TODO add simple work_time_estimator based on WorkUnit.estimate_static
+class DefaultWorkEstimator(WorkTimeEstimator):
+
+    def __init__(self,
+                 rand: Random | None = None):
```
Comment (StannisMod) [inline sampo/schemas/time_estimator.py:41]: default value of `rand` is.. None?

File: sampo/schemas/resources.py [hunk 2]
```diff
@@ -81,12 +81,14 @@ def get_agent_id(self) -> AgentId:
         """
         return self.contractor_id, self.name
 
-    def get_static_productivity(self) -> float:
-        """Return the average productivity of worker team"""
-        return self.productivity.mean * self.count
+    def get_productivity(self, rand: Optional[Random] = None) -> float:
+        """Return the productivity of the worker team
+        It has 2 mods: stochastic and non-stochastic
```
Comment (StannisMod) [inline sampo/schemas/resources.py:86]: This two lines of documentation will be one line without a space

File: sampo/schemas/time_estimator.py [hunk 3]
```diff
@@ -1,21 +1,108 @@
...
+        self._use_idle = True
+        self._mode = WorkEstimationMode.Realistic
+        self.rand = rand
+
+    def find_work_resources(self, work_name: str, work_volume: float) -> dict[str, int]:
+        return {}
```
Comment (StannisMod) [inline sampo/schemas/time_estimator.py:47]: Implement this.

File: sampo/schemas/time_estimator.py [hunk 4]
```diff
@@ -1,21 +1,108 @@
...
+        :param work_unit:
+        :param worker_list:
+        :return: time of task execution
+        """
+        # TODO Is it should be here, not in preprocessing???
+        # TODO Move it to ksg_scheduling
+        # workers = {w.name.replace("_res_fact", ""): w.count for w in worker_list}
+        # work_time = self.estimate_time(work_unit.name.split("_stage_")[0], work_unit.volume, workers)
+        # if work_time > 0:
+        #     return work_time
```
Comment (StannisMod) [inline sampo/schemas/time_estimator.py:66]: commented code

File: tests/scheduler/genetic/fixtures.py [hunk 5]
```diff
@@ -51,7 +51,7 @@ def create_toolbox(wg: WorkGraph,
                    init_schedules: Dict[str, Schedule],
                    rand: Random,
                    spec: ScheduleSpec = ScheduleSpec(),
-                   work_estimator: WorkTimeEstimator = None,
+                   work_estimator: WorkTimeEstimator = DefaultWorkEstimator,
```
Comment (StannisMod) [inline tests/scheduler/genetic/fixtures.py:54]: hmmm

**Review by StannisMod — CHANGES_REQUESTED** [review]

File: sampo/schemas/time_estimator.py (hunk 4 above)
Comment (vanoha) [reply sampo/schemas/time_estimator.py:87]: This code need to be moved to another project

**Pull request merged**